hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -362,7 +356,11 @@ private boolean hasRequiredReplicas(ContainerInfo contInfo) {
     }
     try {
       final ContainerID containerId = ContainerID.valueOf(startContainerID);
-      return scm.getContainerManager().getContainers(containerId, count);
+      if(null == state) {
+        return scm.getContainerManager().getContainers(containerId, count);
+      }
+      return scm.getContainerManager().getContainers(state).stream()
+          .limit(count).collect(Collectors.toList());
```

Comment:
the `startContainerID` does not take effect here.

Reply:
thanks , has been fixed, please take a review

Reply:
You seems miss sort on the stream.

Please check the indent for if and for as well.

Reply:
sorry, fixed